students/Chao/project_fpy/fpy.py

```diff
@@ -0,0 +1,253 @@
+        if self.mgmt.tm.ltm.rules.rule.exists(name=myrule, partition='Common'):
+            result = input("\nExisting iRule {}. Do you want to overwrite? ('y' or 'yes'): ".format(myrule))
+            if result == 'y' or result == 'yes':
+                # Modify the iRule
+                mrule = self.mgmt.tm.ltm.rules.rule.load(name=myrule, partition='Common')
+                mrule.modify(apiAnonymous="when HTTP_REQUEST {\nHTTP::respond 200 content {\n<!DOCTYPE html>\n<title>Site under Maintenance</title>\n}\n}")
+                print("\niRule {} modified.".format(myrule))
+            else:
+                print("\nNo iRule modification was made.")
+        else:
+            # Create new iRule
+            mrule = self.mgmt.tm.ltm.rules.rule.create(name=myrule, partition='Common', apiAnonymous="when HTTP_REQUEST {\nHTTP::respond 200 content {\n<!DOCTYPE html>\n<title>Site under Maintenance</title>\n}\n}")
+            print("\niRule {} created.".format(myrule))
+
+    def check_rule(self, myrule, mypar='Common'):
+        """ Simple function to check if certain iRule already exists """
+        return self.mgmt.tm.ltm.rules.rule.exists(name=myrule, partition=mypar)
+
+
+def display(mgmt):
+    """ Submenu for display menu """
+    while True:
+        print("\n================ Display Menu =================")
+        print("*            (1) Pool Status                  *")
+        print("*            (2) Virtual Server Status        *")
+        print("*            (3) iRule List                   *")
+        print("*            (q) Retuen to Main Menu          *")
+        print("===============================================")
+
+        result = input("Please select a menu item: ")
+
+        if result == '1':
+            p = Pool(mgmt)
+            p.get_pool()
+        elif result == '2':
+            v = Virtual(mgmt)
+            v.get_virtual()
+        elif result == '3':
+            r = Rule(mgmt)
+            r.get_rule()
+        elif result == 'q':
+            break
+        else:
+            print("\n*** Selected item not in the menu. Please try again. ***")
+
+def maintenance(mgmt):
+    """ Submenu for maintenance menu """
+    while True:
+        print("\n================= Maintenance Menu ===================")
+        print("*     (1) Create Maintenance iRule                   *")
+        print("*     (2) Apply Maintenance iRule to All VIPs        *")
+        print("*     (3) Remove Maintenance iRule from All VIPs     *")
+        print("*     (q) Retuen to Main Menu                        *")
+        print("======================================================")
+
+        result = input("Please select a menu item: ")
+
+        if result == '1':
+            r = Rule(mgmt)
+            rule_name = input("Please enter iRule name (optioanl): ")
+            if rule_name:
+                r.create_rule(rule_name)
+            else:
+                r.create_rule()
+        elif result == '2':
```

Comment:
Conditionals 2 and 3 are getting large enough, I would put them in their own function. Actually, they are both the same, except for the words apply vs remove, so you could use one function that flips between apply and remove, either by a dictionary or a conditional.